### PR TITLE
feat: hide/show password functionality on sign-in/up page tm-71

### PR DESCRIPTION
**Pull request by YourTeaChan**
- Password is shown

image

- Password is hidden

image

### CODE REVIEW

**Review by Ann-Maksay — APPROVED** [review]

File: apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx [hunk 1]
```diff
@@ -19,13 +24,19 @@ const SignInForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
 		validationSchema: userSignInValidationSchema,
 	});
 
+	const [isPasswordVisible, setPasswordVisibility] = useState(false);
```
Comment (GEOFARL) [inline apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx:27]: Probably we can extract this logic into a separate hook

File: apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx [hunk 2]
```diff
@@ -19,13 +24,19 @@ const SignInForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
...
 		(event_: React.BaseSyntheticEvent): void => {
 			void handleSubmit(onSubmit)(event_);
 		},
 		[handleSubmit, onSubmit],
 	);
 
+	const handlePasswordVisibility = useCallback(() => {
+		setPasswordVisibility(!isPasswordVisible);
+	}, [isPasswordVisible]);
+
```
Comment (GEOFARL) [inline apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx:39]: Probably we can extract this logic with password visibility into a separate hook

**Review by MappariChan — APPROVED** [review]

File: apps/frontend/src/libs/components/icon-button/icon-button.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { type IconName } from "../icon/libs/types/types.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	className?: string | undefined;
+	iconName: IconName;
+	onClick?: React.MouseEventHandler<HTMLButtonElement>;
+};
+
+const IconButton: React.FC<Properties> = ({
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/icon-button/icon-button.tsx:13]: Use Button component instead. Add a new prop to it `hasVisuallyHiddenLabel`, if it is true we hide the label and show only icon

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/icon-button/icon-button.tsx (hunk 3 above)
Comment (what1s1ove) [reply apps/frontend/src/libs/components/icon-button/icon-button.tsx:13]: agreed

File: apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx (hunk 2 above)
Comment (what1s1ove) [reply apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx:39]: let's do not do it this time. but thanks about suggestion

File: apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx (hunk 1 above)
Comment (what1s1ove) [inline apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx:27]: always add type to useState

File: apps/frontend/src/pages/auth/components/sign-up-form/sign-up-form.tsx [hunk 4]
```diff
@@ -19,13 +24,19 @@ const SignUpForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
 		validationSchema: userSignUpValidationSchema,
 	});
 
+	const [isPasswordVisible, setPasswordVisibility] = useState(false);
```
Comment (what1s1ove) [inline apps/frontend/src/pages/auth/components/sign-up-form/sign-up-form.tsx:27]: same

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/button/button.tsx [hunk 5]
```diff
@@ -36,19 +38,17 @@ const Button: React.FC<Properties> = ({
...
 
-	const icon = iconName ? <Icon name={iconName} /> : null;
-
 	return (
 		<>
 			{href ? (
 				<Link className={buttonStyles} to={href}>
 					{icon}
-					{label}
+					{!hasVisuallyHiddenLabel && label}
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/button/button.tsx:46]: If `hasVisuallyHiddenLabel` is true add `visually-hidden` class, label should always be in the markup

File: apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx [hunk 6]
```diff
@@ -19,13 +19,19 @@ const SignInForm: React.FC<Properties> = ({ onSubmit }: Properties) => {
...
+	const [isPasswordVisible, setPasswordVisibility] = useState<boolean>(false);
+
 	const handleFormSubmit = useCallback(
 		(event_: React.BaseSyntheticEvent): void => {
 			void handleSubmit(onSubmit)(event_);
 		},
 		[handleSubmit, onSubmit],
 	);
 
+	const handlePasswordVisibility = useCallback(() => {
```
Comment (fshabanov) [inline apps/frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx:31]:
```suggestion
	const handleChangePasswordVisibility = useCallback(() => {
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by ShadowOfTheSpace — APPROVED** [review]

**Review by GEOFARL — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**